docker/Dockerfile.sycl

```diff
@@ -116,8 +116,8 @@ RUN SCRATCH_DIR=/scratch && mkdir -p ${SCRATCH_DIR} && cd ${SCRATCH_DIR} && \
     rm -rf ${SCRATCH_DIR}
 
 # Install Kokkos
-ARG KOKKOS_VERSION=3.4.00
-ARG KOKKOS_OPTIONS="-DKokkos_ENABLE_SYCL=ON -DCMAKE_CXX_FLAGS=-Wno-unknown-cuda-version -DKokkos_ENABLE_UNSUPPORTED_ARCHS=ON -DKokkos_ARCH_VOLTA70=ON -DCMAKE_CXX_STANDARD=17"
+ARG KOKKOS_VERSION=bb340781ba774eb0829de03735fe25e89b1176fc
```

Comment:
What version is that?  I presume it is develop branch post 3.5 release.  Please add a comment.

Reply:
Do we want to merge this before the `Kokkos` release?

Reply:
And, yes, this past the Kokkos 3.5 release.

Reply:
3.5.0 did not compile. I originally switched to dea605a1bb5c9a8c1e29a74551fb68c8282528c6 to include kokkos/Kokkos#4358, but that was not enough either.

Reply:
> Do we want to merge this before the Kokkos release?

I would say yes.  Just add something like `# FIXME hash below correspond to 3.5.99 (pre 3.6 release develop branch)`